Core/clim-basic/geometry/regions.lisp

```diff
@@ -465,12 +465,30 @@
 
 ;;; -- 2.5.6 Ellipses and Elliptical Arcs in CLIM ----------------------------
 
+;;; internal protocol
+(defgeneric polar->screen (ellipse)
+  ;; specialized on t, we expect that ellipse protocol is implemented.
+  (:method (ellipse)
+    (nest
+     (multiple-value-bind (rdx1 rdy1 rdx2 rdy2) (ellipse-radii ellipse))
+     (multiple-value-bind (cx cy) (ellipse-center-point* ellipse))
+     (let ((cx (coordinate cx))
+           (cy (coordinate cy))
+           (rdx1 (coordinate rdx1))
+           (rdy1 (coordinate rdy1))
+           (rdx2 (coordinate rdx2))
+           (rdy2 (coordinate rdy2)))
+       (make-3-point-transformation* 0 0 1 0 0 1
+                                     cx cy
+                                     (+ cx rdx1) (+ cy rdy1)
+                                     (+ cx rdx2) (+ cy rdy2))))))
```

Comment:
Why is the method needed? I can't find any subclasses of `ellipse` or `elliptical-arc` besides `standard-ellipse` and `standard-elliptical-arc` respectively and those two don't need the method.

Reply:
quoting a self-imposed rule:
```
;;; - when possible provide a method for a protocol class, and when
;;;   beneficial provide (also) a method for a standard class
```

in principle this method works on any object which implements the ellipse protocol (only protocol functions are invoked). Thanks to that our definitions may be specialized on ellipse/elliptical-arc and it will work for any ellipse (not only standard-ellipse/elliptical-arc).